config.yaml

```diff
@@ -1,6 +1,18 @@
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 options:
+  database_migration_script:
+    type: string
+    description: >-
+      Specifies the relative path from /srv/flask/app that points to a shell script
+      executing database migrations for the Flask application. This script is
+      designed to run once for each Flask container unit. However, users must ensure:
+      1. The script can be executed multiple times without issues;
+      2. Concurrent migrations from different units are safe.
+      In case of migration failure, the charm will re-attempt during the 
```

Comment:
I dont know if this was discussed before: In the case of a failure, why not block the charm so the user can review the script?

Reply:
It's pretty common for the failure to be caused by temporary network issues, so rerun it can be more user-friendly. I think rerun the script might be more "Kubernetes-ic" since that's how Kubernetes treat errors (`CrashLoopBackOff`)